clients/js-legacy/src/actions/amountToUiAmount.ts

```diff
@@ -246,26 +311,38 @@ export async function uiAmountToAmountForMintWithoutSimulation(
 ): Promise<bigint> {
     const accountInfo = await connection.getAccountInfo(mint);
     const programId = accountInfo?.owner;
+
     if (programId !== TOKEN_PROGRAM_ID && programId !== TOKEN_2022_PROGRAM_ID) {
         throw new Error('Invalid program ID');
     }
 
     const mintInfo = unpackMint(mint, accountInfo, programId);
+    const timestamp = await getSysvarClockTimestamp(connection);
```

Comment:
Same here, let's avoid always fetching the clock using whatever technique is best for the other function